### PR TITLE
AGR-2037, AGR-1864, AGR-1823, AGR-1283 code structure for alliance headers, some values are placeho…

### PR DESCRIPTION
**Pull request by azurebrd**
AGR-2037 Headers based on  agr_file_generator repo blob/master/src/generators/header.py and Valerio's code in agr_loader repo.  Strips out blank lines from header (not sure if we should).  Has placeholders for some values, like readme, and species for combined file.  Does not have taxon IDs, but maybe it should.

Could be pulled, but probably should update those variables first.

### CODE REVIEW

**Comment by azurebrd** [comment]
@chris-grove  Recent files created by this update : 
http://download.alliancegenome.org/3.1.0/INTERACTION-MOL/COMBINED/INTERACTION-MOL_COMBINED_4.tar.gz
http://download.alliancegenome.org/3.1.0/INTERACTION-GEN/COMBINED/INTERACTION-GEN_COMBINED_10.tar.gz

**Comment by azurebrd** [comment]
(Oh, plan to remove the commented out old headers once the new ones are approved, but leaving them in for easy of bringing them back if needed)

**Comment by chris-grove** [comment]
@azurebrd 
Thanks! I see the headers:

```
##########################################################################
#
# Data type: Molecular Interactions
# Data format: PSI-MI TAB 2.7 Format
# README: Readme placeholder
# Source: Alliance of Genome Resources (Alliance)
# Source URL: http://alliancegenome.org/downloads
# Help Desk: help@alliancegenome.org
# Species: Many
# Alliance Database Version: 3.1.0
# File generated (UTC): 2020-05-30 23:44
#
##########################################################################



##########################################################################
#
# Data type: Genetic Interactions
# Data format: PSI-MI TAB 2.7 Format
# README: Readme placeholder
# Source: Alliance of Genome Resources (Alliance)
# Source URL: http://alliancegenome.org/downloads
# Help Desk: help@alliancegenome.org
# Species: Many
# Alliance Database Version: 3.1.0
# File generated (UTC): 2020-05-30 23:41
#
##########################################################################
```

but I'd like the COMBINED files to be like:

```
##########################################################################
#
# Data type: Molecular Interactions
# Data format: PSI-MI TAB 2.7 Format
# README: https://github.com/HUPO-PSI/miTab/blob/master/PSI-MITAB27Format.md
# Source: Alliance of Genome Resources (Alliance)
# Source URL: http://alliancegenome.org/downloads
# Help Desk: help@alliancegenome.org
# TaxonIDs: NCBI:txid10090, NCBI:txid9606, NCBI:txid7955, NCBI:txid10116, NCBI:txid7227, NCBI:txid6239, NCBI:txid559292
# Species: Mus musculus, Homo sapiens, Danio rerio, Rattus norvegicus, Drosophila melanogaster, Caenorhabditis elegans, Saccharomyces cerevisiae
# Alliance Database Version: 3.1.0
# File generated (UTC): 2020-05-30 23:44
#
##########################################################################



##########################################################################
#
# Data type: Genetic Interactions
# Data format: PSI-MI TAB 2.7 Format
# README: https://github.com/HUPO-PSI/miTab/blob/master/PSI-MITAB27Format.md
# Source: Alliance of Genome Resources (Alliance)
# Source URL: http://alliancegenome.org/downloads
# Help Desk: help@alliancegenome.org
# TaxonIDs: NCBI:txid10090, NCBI:txid9606, NCBI:txid7955, NCBI:txid10116, NCBI:txid7227, NCBI:txid6239, NCBI:txid559292
# Species: Mus musculus, Homo sapiens, Danio rerio, Rattus norvegicus, Drosophila melanogaster, Caenorhabditis elegans, Saccharomyces cerevisiae
# Alliance Database Version: 3.1.0
# File generated (UTC): 2020-05-30 23:41
#
##########################################################################
```

Single species files will just have a single taxon ID and species name.

**Comment by azurebrd** [comment]
AGR-2037 standardized headers based on agr_file_generator repo
AGR-1864 biogrid molecular interaction protein-RNA data in
AGR-1823 non-pubmed ID paper references allowed, allowing DOIs among others
Latest files uploaded by this code 
  http://download.alliancegenome.org/3.1.0/INTERACTION-GEN/COMBINED/INTERACTION-GEN_COMBINED_11.tar.gz
  http://download.alliancegenome.org/3.1.0/INTERACTION-MOL/COMBINED/INTERACTION-MOL_COMBINED_5.tar.gz
(as well as non-COMBINED dataSubTypes)

**Comment by christabone** [comment]
Just updating the title to associate the other tickets with this PR.

File: src/default_env_vars.yml
Comment (christabone) [inline src/default_env_vars.yml:9]: Will this go somewhere more permanent at some point? Just curious?

**Review by christabone — APPROVED** [review]

File: src/default_env_vars.yml
Comment (azurebrd) [reply src/default_env_vars.yml:9]: Valerio's just made these changes https://github.com/alliance-genome/agr_file_generator/pull/85 to create a pip module for the agr_file_generator repo, so that we can generate the headers with it.  I may add this to preprocess for 3.1, but the headers should already be in the same format, so I may wait until the next release, since we'll have to talk to people and curators about how the headers should be formed.  (Right now if the headers format changes slightly, we'd just need to change one variable in the .yml file, if we change it to the module, I think we'd have to change the requirements file + the imports in the .py files that use it.  If the headers change a lot the module would be better though)

**Pull request merged**
